src/main/scala/cromwell/engine/package.scala

```diff
@@ -34,4 +34,9 @@ package object engine {
     override def toString: String = "Succeeded"
     override val isTerminal = true
   }
+
+  trait WorkflowDescriptor {
+    def name: String
+    def id: WorkflowId
+  }
```

Comment:
I think it's semantics mostly... what describes a workflow?  certainly the WorkflowId.  But name might not be enough to uniquely identify one and only one workflow.  I'd vote for having the trait have "namespace" and "id"